core/unit_test/TestViewPadding.hpp

```diff
@@ -0,0 +1,767 @@
+  result &=
+      padding_subview_and_copy<DType_7, Kokkos::Impl::ViewDimension<ARG_7>, ll,
+                               no_pad>(RANK_7, PARAM_7);
+  ASSERT_TRUE(result);
+  using lr = Kokkos::LayoutRight;
+  result &=
+      padding_subview_and_copy<DType_1, Kokkos::Impl::ViewDimension<ARG_1>, lr,
+                               pad>(RANK_1, PARAM_1);
+  result &=
+      padding_subview_and_copy<DType_2, Kokkos::Impl::ViewDimension<ARG_2>, lr,
+                               pad>(RANK_2, PARAM_2);
+  result &=
+      padding_subview_and_copy<DType_3, Kokkos::Impl::ViewDimension<ARG_3>, lr,
+                               pad>(RANK_3, PARAM_3);
+
+  result &=
+      padding_subview_and_copy<DType_4, Kokkos::Impl::ViewDimension<ARG_4>, lr,
+                               pad>(RANK_4, PARAM_4);
+
+  result &=
+      padding_subview_and_copy<DType_5, Kokkos::Impl::ViewDimension<ARG_5>, lr,
+                               pad>(RANK_5, PARAM_5);
+
+  result &=
+      padding_subview_and_copy<DType_6, Kokkos::Impl::ViewDimension<ARG_6>, lr,
+                               pad>(RANK_6, PARAM_6);
+  ASSERT_TRUE(result);
+  result &=
+      padding_subview_and_copy<DType_7, Kokkos::Impl::ViewDimension<ARG_7>, lr,
+                               pad>(RANK_7, PARAM_7);
+  result &=
+      padding_subview_and_copy<DType_1, Kokkos::Impl::ViewDimension<ARG_1>, lr,
+                               no_pad>(RANK_1, PARAM_1);
+  result &=
+      padding_subview_and_copy<DType_2, Kokkos::Impl::ViewDimension<ARG_2>, lr,
+                               no_pad>(RANK_2, PARAM_2);
+  result &=
+      padding_subview_and_copy<DType_3, Kokkos::Impl::ViewDimension<ARG_3>, lr,
+                               no_pad>(RANK_3, PARAM_3);
+  result &=
+      padding_subview_and_copy<DType_4, Kokkos::Impl::ViewDimension<ARG_4>, lr,
+                               no_pad>(RANK_4, PARAM_4);
+  result &=
+      padding_subview_and_copy<DType_5, Kokkos::Impl::ViewDimension<ARG_5>, lr,
+                               no_pad>(RANK_5, PARAM_5);
+  result &=
+      padding_subview_and_copy<DType_6, Kokkos::Impl::ViewDimension<ARG_6>, lr,
+                               no_pad>(RANK_6, PARAM_6);
+  result &=
+      padding_subview_and_copy<DType_7, Kokkos::Impl::ViewDimension<ARG_7>, lr,
+                               no_pad>(RANK_7, PARAM_7);
+  ASSERT_TRUE(result);
+
+  // Padding is not allowed for LayoutStride
+  // using ls = Kokkos::LayoutStride;
+}
+
+#undef PARAM_0
+#undef PARAM_1
+#undef PARAM_2
+#undef PARAM_3
+#undef PARAM_4
+#undef PARAM_5
+#undef PARAM_6
+#undef PARAM_7
+
+#undef PARAM_Nz_0
+#undef PARAM_Nz_1
+#undef PARAM_Nz_2
+#undef PARAM_Nz_3
+#undef PARAM_Nz_4
+#undef PARAM_Nz_5
+#undef PARAM_Nz_6
+#undef PARAM_Nz_7
+
+#undef PARAM_0_RANK
+#undef PARAM_1_RANK
+#undef PARAM_2_RANK
+#undef PARAM_3_RANK
+#undef PARAM_4_RANK
+#undef PARAM_5_RANK
+#undef PARAM_6_RANK
+#undef PARAM_7_RANK
+
+#undef SUBRANGE_DIM
+
+#undef UNUSED
+
+#undef DType
```

Comment:
This is not a defined macro.

There are missing undefines for `ZERO` and `ARG_*`